rclpy/rclpy/node.py

```diff
@@ -761,6 +763,25 @@ def _set_parameters_atomically(
 
         return result
 
+    def add_on_set_paramters_callback(
+        self,
+        callback: Callable[[List[Parameter]], SetParametersResult]
+    ) -> None:
+        """Add the callback to list."""
+        prevCallBack = self._parameters_callback
```

Comment:
Storing the list in `prevCallBack` temporarily and later assigning it back to `self._parameters_callback` doesn't make any sense. `insert()` modifies the list in place so these two lines are no-ops.